esp/services/ws_codesign/ws_codesignService.cpp

```diff
@@ -90,3 +90,34 @@ bool Cws_codesignEx::onListUserIDs(IEspContext &context, IEspListUserIDsRequest
     resp.setUserIDs(userIds);
     return true;
 }
+
+bool Cws_codesignEx::onVerify(IEspContext &context, IEspVerifyRequest &req, IEspVerifyResponse &resp)
+{
+    const char* text = req.getText();
+    if (!text || !*text)
+    {
+        resp.setErrMsg("No text provided");
+        return false;
+    }
+    StringBuffer signer;
+    bool isValidSig = false;
+
+    try
+    {
+        isValidSig = queryCodeSigner().verifySignature(text, signer);
+    }
+    catch (IException *e)
+    {
+        StringBuffer msg;
+        e->errorMessage(msg);
+        auto code = e->errorCode();
+        e->Release();
+        OERRLOG("Signature verify error %d: %s", code, msg.str());
+        resp.setIsVerified(false);
```

Comment:
I think it's good that you aren't putting internal details in resp.setErrMsg(""), but wonder if a generic message would be good.  Like "Signature verify error"?

Reply:
> @shamser curious whether you tested the secrets coming from a vault?

Yes, I've done this ticket: https://track.hpccsystems.com/browse/HPCC-25073.